plugins/eos/gs-plugin-eos.c

```diff
@@ -2290,6 +2291,62 @@ gs_plugin_app_upgrade_download (GsPlugin *plugin,
 	return TRUE;
 }
 
+static char *
+get_os_collection_id (void)
+{
+	OstreeDeployment *booted_deployment;
+	GKeyFile *origin;
+	g_autofree char *refspec = NULL;
+	g_autofree char *remote = NULL;
+	g_autofree char *collection_id = NULL;
+	g_autoptr(OstreeRepo) repo = NULL;
+	g_autoptr(OstreeSysroot) sysroot = NULL;
+	g_autoptr(GError) error = NULL;
+
+	sysroot = ostree_sysroot_new_default ();
+	if (!ostree_sysroot_load (sysroot, NULL, NULL))
+		return NULL;
+
+	booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
+	if (booted_deployment == NULL)
+		return NULL;
+
+	origin = ostree_deployment_get_origin (booted_deployment);
+	if (origin == NULL)
+		return NULL;
+
+	refspec = g_key_file_get_string (origin, "origin", "refspec", NULL);
+	if (refspec == NULL)
+		return NULL;
+
+	ostree_parse_refspec (refspec, &remote, NULL, NULL);
+	if (remote == NULL)
+		return NULL;
+
+	repo = ostree_repo_new_default ();
+
+	if (!ostree_repo_open (repo, NULL, NULL))
+		return NULL;
+
+	ostree_repo_get_remote_option (repo, remote, "collection-id", NULL, &collection_id, &error);
```

Comment:
Nitpick: `error` is unused here.

Reply:
Locally, I filled in all the potential uses of `error` in this function.

Reply:
Could use an `if (error != NULL) goto err_out;` here